mobile/client.go

```diff
@@ -163,8 +170,29 @@ func (os *Objects) Get(i int) *Props { return os.a[i] }
 func (os *Objects) Length() int      { return len(os.a) }
 func (os *Objects) String() string   { return fmt.Sprint(os.a) }
 
+type PropsUtil struct {
+}
+
+func (pu PropsUtil) JSONString(props Props) string {
+	if props.m == nil {
+		return ""
+	}
+	if bt, err := props.MarshalJSON(); err == nil {
+		return string(bt)
+	}
+	return ""
+}
+func (pu PropsUtil) FromJsonString(propsString string) Props {
+	var props Props
+	if err := props.UnMarshalJSON([]byte(propsString)); err != nil {
+		return Props{}
+	}
+	return props
+}
+
 type PropVisitor interface {
 	VisitBool(string, bool) error
+	VisitInt(string, int) error
```

Comment:
This breaks backward-compatibility. How about simply calling VisitNumber in case of an int?

Reply:
Sorry, did not get you. VisitNumber only support `float64`. Should I cust the int to float64 to use VisitNumber?

It seems if I put int value in the props, the visit props never call for this value(int) because there were not int case in the switch statement.

Reply:
Since JSON doesn't retain type information, the client code needs to be prepared to receive numbers as float64, which is the type which the JSON decoder falls back to. Casting ints to float64 should be an acceptable solution here, as it will happen in other situations anyway.

Reply:
> the client code needs to be prepared to receive numbers as float64, which is the type which the JSON decoder falls back to

I think this will be the expected behavior. I understood why `VisitInt` is available. I think I will remove it. It's it redundant.

Reply:
What do you mean with "VisitInt is available"? It wasn't until you added it in this PR. :smile: Adding it would require a major version bump (at least for ninchat-go which is at version 1.4), as PropVisitor implementations would break.

Do you mean that there are SetInt and GetInt accessors? GetInt is feasible as it knows that an int is wanted, and thus it can convert the float64 if it finds one. With the visitor pattern, it's not possible to know what type the visitor wants; the visitor needs to do that conversion in its own code if it knows that a particular property is really an int.

Reply:
Sorry for the confusion. I meant, I added it un-necessarily. I should remove that in the PR. 🗡️  😸